include/wx/valnum.h

```diff
@@ -184,6 +184,14 @@ class wxNumValidator : public B
         max = GetMax();
     }
 
+    virtual bool IsInRange(LongestValueType value) const wxOVERRIDE
+    {
+        wxGCC_WARNING_SUPPRESS(sign-compare)
+        const bool isInRange = GetMin() <= value && value <= GetMax();
+        wxGCC_WARNING_RESTORE(sign-compare)
+        return isInRange;
```

Comment:
Couldn't this be just
```suggestion
        const ValueType valueT = static_cast<ValueType>(value);
        return GetMin() <= valueT && valueT <= GetMax();
```
?

Reply:
With [`ValueType == unsigned`] --> (`ULONG_MAX == 4294967295UL`)
we expect these tests to succeed:
```
    m_text->ChangeValue("-234");
    CPPUNIT_ASSERT( !valUnsigned.TransferFromWindow() );

    CPPUNIT_ASSERT( !valUnsigned.IsInRange(4294967296LL) );
```
but they fail! because `IsInRange()` will return true for both values:
```
static_cast<unsigned>(-234) == 4294967062; // 0 <= 4294967062 && 4294967062 <= ULONG_MAX
static_cast<unsigned>(4294967296LL) == 0; // 0 <= 0 && 0 <= ULONG_MAX
```
The first is simple to fix by adding this to `TransferFromWindow()`:
```
...
            if ( s.empty() && BaseValidator::HasFlag(wxNUM_VAL_ZERO_AS_BLANK) )
                value = 0;
            else if ( GetMin() >= 0 && s.StartsWith("-") )
                return false;
            else if ( !BaseValidator::FromString(s, &value) )
                return false;
...
```

but i don't know how to fix the second one ?

Reply:
I see, thanks. I'll try fixing this tomorrow, I believe we just need to stop using `LongestValueType` hack completely, I don't understand what do we gain from it, while we clearly lose a lot, both in code correctness and in code clarity.